From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Richard Tollerton <rich.tollerton@ni.com>
Date: Wed, 30 Sep 2020 19:13:44 -0500
Subject: [PATCH] chkdecimalpoint: use default libs

Just about all of the libtest/ tests link fine if you just always link against
libcurl, with the sole exception of chkdecimalpoint, because of duplicate
symbols caused by explicitly including lib/mprintf.c. This makes integration
into independent downstream build systems more complicated. Work around this by
removing all of the ../../lib/* sources from chkdecimalpoint and letting LDADD
float to allow normal curl objects to be linked in.

Upstream-Status: Inappropriate [workarounds for wacky builds]

diff --git a/tests/libtest/Makefile.inc b/tests/libtest/Makefile.inc
index 5618952..659322e 100644
--- a/tests/libtest/Makefile.inc
+++ b/tests/libtest/Makefile.inc
@@ -63,9 +63,7 @@ noinst_PROGRAMS = chkhostname libauthretry libntlmconnect                \
          lib1915 lib1916 lib1917 lib1918 lib1933 lib1934 lib1935 lib1936 \
          lib3010
 
-chkdecimalpoint_SOURCES = chkdecimalpoint.c ../../lib/mprintf.c \
- ../../lib/curl_ctype.c  ../../lib/dynbuf.c ../../lib/strdup.c
-chkdecimalpoint_LDADD =
+chkdecimalpoint_SOURCES = chkdecimalpoint.c
 chkdecimalpoint_CPPFLAGS = $(AM_CPPFLAGS) -DCURL_STATICLIB \
  -DCURLX_NO_MEMORY_CALLBACKS -DBUILDING_LIBCURL
 
-- 
2.25.1

